In [1]:
""" Linear regression with SciPy """
from scipy import stats

First let's just assume some values for stock returns and market returns. In other words $E(r_i)-r_f$ and $E(r_m)-r_f$.

In reality you would get these from real data, but this is just for illustration.

In [2]:
stock_returns = [0.065, 0.0265, -0.0593, -0.001, 0.0346]
mkt_returns = [0.055, -0.09, -0.041, 0.045, 0.022]

In [4]:
beta, alpha, r_value, p_value, std_err = stats.linregress(stock_returns, mkt_returns)

print beta, alpha, p_value

0.507743187877 -0.00848190035246 0.522715523909


So the $\beta$ of the stock is $0.5077$.

The equation describing the securities market line (SML) is
\begin{align}
E(r_i) = r_f + \beta_i [ E(r_m) - r_f]
\end{align}

The following screen dump reminds us of the SML from class. Please notice the notation is slightly different.
<img src="SML.png">

Therefore if we plug in some values on the right-hand side we get the expected return on the left-hand side. So assume $r_f=5\%$ and the market risk premium is $[ E(r_m) - r_f] = 8.5\%$ then the formula becomes
\begin{align}
5 + 0.5077 \times 8.5\% = 9.3\%
\end{align}
where $0.5077 \times 8.5 = 4.3\%$ is the risk premium of an equity with $\beta=0.5077$, and its expected return is $9.3\%$.

The following cell uses code to compute these outcomes.

In [6]:
""" Calculating the SML """
rf = 0.05
mrisk_prem = 0.085
risk_prem = mrisk_prem * beta
print risk_prem

0.0431581709696


In [7]:
expected_stock_return = rf + risk_prem
print expected_stock_return

0.0931581709696


## Undervaluation and Overvaluation

So the expected return is $9.3\%$ but if it actually returned $10.5\%$ we would say that the SML undervalues the stock, whereas if it actually returned $7\%$ we would say the SML overvalues the security. So above the line is stocks are undervalued by the SML and below the line they are undervalued by the SML.